tests/publiccloud/cloud_netconfig.pm

```diff
@@ -50,10 +50,25 @@ sub run {
       $instance->ssh_script_output(cmd => $grep_cloudvm_ipv4_cmd);
     chomp($local_eth0_ip);
 
-    my $query_meta_ipv4_cmd =
-      'curl -H Metadata:true '
-      . '"http://169.254.169.254/metadata/instance/network/interface/0/ipv4/'
-      . 'ipAddress/0/privateIpAddress?api-version=2023-07-01&format=text"';
+    my $query_meta_ipv4_cmd = "";
+    if (check_var('PUBLIC_CLOUD_PROVIDER', 'AZURE')) {
+        $query_meta_ipv4_cmd =
+          'curl -H Metadata:true '
+          . '"http://169.254.169.254/metadata/instance/network/interface/0/ipv4/'
+          . 'ipAddress/0/privateIpAddress?api-version=2023-07-01&format=text"';
+    }
+    elsif (check_var('PUBLIC_CLOUD_PROVIDER', 'EC2')) {
+        $query_meta_ipv4_cmd =
+          'curl "http://169.254.169.254/latest/meta-data/local-ipv4"';
+    }
+    elsif (check_var('PUBLIC_CLOUD_PROVIDER', 'GCE')) {
+        $query_meta_ipv4_cmd =
+          'curl -H "Metadata-Flavor: Google" '
+          . '"http://169.254.169.254/computeMetadata/v1/instance/'
+          . 'network-interfaces/0/ip"';
```

Comment:
What is this magical link? I think it deserves a comment at least

Reply:
also it would be nice to store the common part in a variable as it is re-used in multiple places

Reply:
This is the metadata service of the cloud provider. In case of GCE it is this one, in case of other providers it is different. What is always the same is the IP it is provided under: `169.254.169.254`.

I can add a comment if it is really not clear. However, I thought it is clear if one is aware of the cloud metadata service.

Reply:
> also it would be nice to store the common part in a variable as it is re-used in multiple places
Maybe it would indeed make it more clear if querying the metadata service was a library function.

I am going to have a look if I can improve the clarity.

Reply:
I believe there are native commands in the cloud images to access this metadata without using curl.  Or at least that's the way things should be.

Reply:
Well, it is `cloud-netconfig`s job to assign the data which the metadata service provides to the instance. For this reason, I verify that this information is the same like the one in `ip`.

Since it is exactly not the job of `cloud-netconfig` to use data from the public cloud API, it doesn't seem as the right place to compare with from my perspective.

Reply:
Here is an example of fix on `cloud-netconfig` side about that API https://github.com/SUSE-Enceladus/cloud-netconfig/pull/24

Reply:
> I mean the usual aws, az & gcloud cloud-native commands.

I don't think so , commands which you mention they designed for **consumers** of Public Cloud. On other hand the data which is provided by this metadata server is **backend** data which is relevant only for AWS/Azure/Google employees who maintain cloud infra AND for guys like us who maintain images for this clouds. So I don't know for sure but I will be surprised if you may get this info from those commands . 

Also I was trying to find domain name for this IP but look what I found instead https://www.whatismyip.com/169-254-ip-address/ . Seems like it is not just random IP , but there is some common agreement :thinking:  I will try to get more info from developers

Reply:
Thank you for the comments. I think I understand what you meant.
As a result, I have moved the IP to variable and described it with a comment.